pyccel/parser/semantic.py

```diff
@@ -362,12 +362,19 @@ def insert_class(self, cls, parent=False):
         else:
             raise TypeError('Expected A class definition ')
 
+    def insert_template(self, expr):
+        """insert a template"""
```

Comment:
What are we inserting a template into?

Reply:
they are inserted to the templates of the scope

Reply:
The templates are inserted into the scope? Or something is inserted into the templates of the scope?
Whichever it is can you write it in the docstring please

Reply:
a scope have a list of templates, here we append the given template to that list

Reply:
yes sure